ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -158,18 +205,37 @@ def create_input_processors(
         :param unnormalized_inputs: Vector inputs that should not be normalized, and added to the vector
             obs.
         :param normalize: Normalize all vector inputs.
-        :return: Tuple of visual encoders and vector encoders each as a list.
+        :return: Tuple of :
+         - ModuleList of the encoders (None if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of embedding sizes (0 if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of the inputs that need to be processed by a variable length observation encder.
         """
         encoders: List[nn.Module] = []
         embedding_sizes: List[int] = []
-        for obs_spec in observation_specs:
-            encoder, embedding_size = ModelUtils.get_encoder_for_obs(
-                obs_spec.shape, normalize, h_size, vis_encode_type
-            )
-            encoders.append(encoder)
-            embedding_sizes.append(embedding_size)
+        var_len_indices: List[int] = []
+        for idx, obs_spec in enumerate(observation_specs):
+            if ModelUtils.can_encode_attention(obs_spec):
+                # This is a 2D tensor
+                # TODO : better if condition
+                var_len_indices.append(idx)
+                encoders.append(None)
+                embedding_sizes.append(0)
+            elif ModelUtils.can_encode_vector(obs_spec) or ModelUtils.can_encode_visual(
```

Comment:
I think these `can_encode_x` functions should be more forceful i.e. `encodes_x` and only one can be true for a given obs i.e. only one of encodes_attention, encodes_visual, encodes_vector can be true.

Reply:
Would it be better if I introduced an enum to do that and have a single method like "get encoder type" ? I can try that.